Tests/test_image_convert.py

```diff
@@ -123,6 +123,26 @@ def test_trns_RGB(self):
         self.assertNotIn('transparency', p.info)
         p.save(f)
 
+    # ref https://github.com/python-pillow/Pillow/issues/1979
+
+    def test_p_la(self):
+        def L(rgb):
+            return int((rgb[0] * 299 + rgb[1] * 587 + rgb[2] * 114) / 1000)
```

Comment:
You just need to check for alpha channel, right? No need to check conversion to L itsef.